.github/workflows/blocks-playwright.yml

```diff
@@ -58,19 +69,74 @@ jobs:
 
+            - name: Archive flaky test reports
+              uses: actions/upload-artifact@v4
+              if: ${{ !cancelled() }}
+              with:
+                  name: flaky-tests-report-shard-${{ matrix.shardIndex }}
+                  path: plugins/woocommerce-blocks/flaky-tests
+                  if-no-files-found: ignore
+
     merge-artifacts:
         # Merges all artifacts from all shards into a single zip and
         # deletes the parts. In case of a rerun, artifacts from the
-        # previous run will be retained by merging them with the new ones. 
+        # previous run will be retained by merging them with the new ones.
         name: Merge Artifacts
         if: ${{ !cancelled() }}
         needs: [blocks-playwright-tests]
         runs-on: ubuntu-latest
+        outputs:
+            has-flaky-test-report: ${{ !!steps.merge-flaky-tests-reports.outputs.artifact-id }}
         steps:
-            - name: Merge Artifacts
+            - name: Merge failures artifacts
               uses: actions/upload-artifact/merge@v4
               # Don't fail the job if there aren't any artifacts to merge.
               continue-on-error: true
               with:
                   name: failures-artifacts
+                  pattern: failures-artifacts*
                   delete-merged: true
+
+            - name: Merge flaky tests reports
+              id: merge-flaky-tests-reports
+              uses: actions/upload-artifact/merge@v4
+              continue-on-error: true
+              with:
+                  name: flaky-tests-report
+                  pattern: flaky-tests-report*
+                  delete-merged: true
+
+    create-github-issues-for-flaky-tests:
+        name: Create GitHub issues for flaky tests
+        needs: [merge-artifacts]
+        if: ${{ !cancelled() && needs.merge-artifacts.outputs.has-flaky-test-report == 'true' }}
+        runs-on: ubuntu-latest
+        permissions:
+            contents: read
+            issues: write
+        steps:
+            # We are using @wordpress/report-flaky-tests package from Gutenberg repo
+            # to create GitHub issues for flaky tests.
+            - uses: actions/checkout@v4
+              with:
+                  repository: WordPress/gutenberg
+                  ref: trunk
+
+            - uses: actions/download-artifact@v4
+              with:
+                  name: flaky-tests-report
+                  path: flaky-tests
+
+            - name: Setup Node.js and install dependencies
+              uses: ./.github/setup-node
+
+            - name: Npm build
+              # TODO: We don't have to build the entire project, just the action itself.
+              run: npm run build:packages
+
+            - name: Report flaky tests
+              uses: ./packages/report-flaky-tests
+              with:
+                  repo-token: '${{ secrets.GITHUB_TOKEN }}'
+                  label: 'metric: flaky e2e test'
```

Comment:
I have added the label 'metric: flaky e2e test' for flaky tests. 

**Question:** Should we consider adding any other label too? For example: 
- focus: FSE
- Team label: team: Kirigami & Origami
	- If we decide to add a team label, then I believe we will need to add "team: Rubik" as we both work on code covered by this GitHub Action i.e. `'plugins/woocommerce-blocks/**'`

Reply:
Adding extra labels would be great.
It would help for now differentiate these issues from those raised by the core e2e suite.
Team labels would be ideal if possible as this means they wouldn't show us as unassigned in issue triage (along we could just tweak the triage query to exclude them as another option).

Reply:
Yeah, it makes sense to add extra labels. Will adding "focus: FSE" label be enough? Or should we also add team labels?

I think the problem with adding team labels is that we will need to add both of the following labels:
1. `team: Kirigami & Origami`
2. `team: Rubik`

Now, as part of the triage process, we will need to remove one of these because the flaky test will be related to only one team. I am also going to discuss it internally.

Reply:
JFYI I asked the same question on Slack too - p1717406864428249-slack-C02FL3X7KR6

Reply:
Update: We decided to add both team labels with issues that will be created by this Flaky Test Reporter.